src/main/java/org/glite/authz/pep/pip/provider/PolicyNamesPIP.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) Members of the EGEE Collaboration. 2006-2010.
+ * See http://www.eu-egee.org/partners/ for details on the copyright holders.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors:
+ * 2016-
+ *    Mischa Salle <msalle@nikhef.nl>
+ *    Rens Visser <rensv@nikhef.nl>
+ *    NIKHEF Amsterdam, the Netherlands
+ *    <grid-mw-security@nikhef.nl>
+ */
+
+package org.glite.authz.pep.pip.provider;
+
+import org.glite.authz.pep.pip.provider.policynamespip.UpdatingPolicyNamesCache;
+
+import org.glite.authz.common.model.Request;
+import org.glite.authz.common.model.Subject;
+import org.glite.authz.common.model.Attribute;
+
+import org.glite.authz.pep.pip.PIPProcessingException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.io.IOException;
+import java.text.ParseException;
+
+
+/**
+ * This PIP searches for all appearances of the issuer of the end-entity
+ * certificate in the IGTF .info files. The resulting set is pushed into a
+ * {@value #ATTR_CA_POLICY_NAMES} attribute.
+ * @author Mischa Sall&eacute;, Rens Visser 
+ */
+public class PolicyNamesPIP extends AbstractPolicyInformationPoint {
+    /** Class logger instance */
+    private final Logger log = LoggerFactory.getLogger(PolicyNamesPIP.class);
```

Comment:
I also believe there is only one PIP instance, although you probably now better. It's fine with me to make it static and final (I will), but would then strongly suggest we do the same for all the other PIPs currently in the PEPd code base.